frontend/src/poc-code/testK8s/TestK8s.tsx

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable no-console */
+import * as React from 'react';
+import { Button, PageSection, TextInput } from '@patternfly/react-core';
+import {
+  k8sCreateResource,
+  k8sDeleteResource,
+  k8sGetResource,
+  k8sListResource,
+  K8sModel,
+  k8sPatchResource,
+  K8sResourceCommon,
+  k8sUpdateResource,
+  // useK8sWatchResource,
+} from './dynamic-plugin-sdk';
+
+const ProjectModel: K8sModel = {
+  apiVersion: 'v1',
+  apiGroup: 'project.openshift.io',
+  kind: 'Project',
+  abbr: 'PR',
+  label: 'Project',
+  labelPlural: 'Projects',
+  plural: 'projects',
+};
+const ConfigMapModel: K8sModel = {
+  apiVersion: 'v1',
+  kind: 'ConfigMap',
+  abbr: 'CM',
+  plural: 'configmaps',
+  labelPlural: 'ConfigMaps',
+  label: 'ConfigMap',
+  namespaced: true,
+};
+
+// eslint-disable-next-line no-shadow
+enum ActionType {
+  LIST = 'list',
+  CREATE = 'create',
+  GET = 'get',
+  PATCH = 'patch',
+  PUT = 'put',
+  DELETE = 'delete',
+}
+
+const TestK8s: React.FC = () => {
+  const [r, setR] = React.useState(null);
+  const [namespace, setNamespace] = React.useState<string>('default');
+  const [name, setName] = React.useState<string>('test');
+  const [status, setStatus] = React.useState<string>('');
+  const [action, setAction] = React.useState<ActionType | null>(null);
+
+  // const result = useK8sWatchResource({
+  //   groupVersionKind: { version: 'v1', kind: 'ConfigMap' },
+  //   name: 'test',
+  //   namespace,
+  // });
+  // const result = useK8sWatchResource(
+  //   namespace === 'default' // default namespace has 100s of configmaps, don't fetch it
+  //     ? {}
+  //     : {
+  //         // groupVersionKind: { group: 'project.openshift.io', version: 'v1', kind: 'Project' },
+  //         groupVersionKind: { version: 'v1', kind: 'ConfigMap' },
+  //         isList: true,
+  //         namespace,
+  //       },
+  // );
+  // const hookResource = result[0];
+  // console.debug('render result', result);
+
+  React.useEffect(() => {
+    const testConfigMapMetadata = {
+      metadata: {
+        name,
+        namespace,
+      },
+    };
+    const testConfigMapData: K8sResourceCommon & { [key: string]: any } = {
+      apiVersion: ConfigMapModel.apiVersion,
+      kind: ConfigMapModel.kind,
+      ...testConfigMapMetadata,
+      data: {
+        test: 'true',
+      },
+    };
```

Comment:
This constant can be moved outside of the component and only enhanced with `testConfigMapMetadata`